### ISSUE TITLE
Fix false positives for single-line comments in unit-no-unknown and number-no-trailing-zeros

### ISSUE THREAD

**Issue by DemianX0** [stylelint/stylelint#4616]
> Clearly describe the bug

There are false positives inside single-line comments for "9+", "4.0" (any number).

> Which rule, if any, is the bug related to?

single-line comments
`unit-no-unknown`
`number-no-trailing-zeros`

Related: #4617, [#4298](https://github.com/stylelint/stylelint/issues/4298#issuecomment-590111620), #4049, #3975

> What code is needed to reproduce the bug?

[Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACAzFBaAzgQwHafgB1cACE9bAB0oBsYsBPTAFxgFsAaEgem5IE4A1MTIByAEoQARhGYRRXXiQCCuKACcIASygkALADoADCJI58WGOq0oA3MRAcQmBrmbYAHghB1MmR+AQuChaAObewKZkhCDqAK6+MfAkkbiEzDFxuFrM6LgQ6FkA1vkA7mkIJMzxMBzE6TG4cWxSVnkF1dhaNFq4oegAXlYQmElVNXWkZNPRztW9-QCOcbIwo5UxmAt0MVEkAL7E+yD7QA)

```less
@ff-sans:
  -apple-system, // 9+
  'Roboto', // Android 4.0
  sans-serif;
```

> What stylelint configuration is needed to reproduce the bug?

```json
	"unit-no-unknown": true,
	"number-no-trailing-zeros": true,
```

> Which version of stylelint are you using?

`13.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`grunt stylelint`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Less. No single-line comments in CSS.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:
```
3:26 Unexpected trailing zero(s) (number-no-trailing-zeros)
2:21 Unexpected unknown unit "+" (unit-no-unknown)
```

**Comment by DemianX0** [comment]
Note that in a 200+ line .less the false positives were attributed to the wrong line numbers. Usually, the line number was 1-3 more than what triggered the rule.

**Comment by jeddy3** [comment]
Thanks for the report and for using the template.

Please consider [contributing to the upstream syntax](https://stylelint.io/developer-guide/syntaxes) to stylelint [uses to support Less](https://stylelint.io/about/syntaxes).

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider fixing the issue upstream as the syntax is no longer bundled by Stylelint.
